internal/command/action_v2_target_model.go

```diff
@@ -115,6 +116,54 @@ func (wm *TargetWriteModel) NewChangedEvent(
 	return target.NewChangedEvent(ctx, agg, changes)
 }
 
+type TargetsExistsWriteModel struct {
+	eventstore.WriteModel
+	ids         []string
+	existingIDs []string
+}
+
+func (e *TargetsExistsWriteModel) AllExists() bool {
+	return len(e.ids) == len(e.existingIDs)
+}
+
+func NewTargetsExistsWriteModel(ids []string, resourceOwner string) *TargetsExistsWriteModel {
+	return &TargetsExistsWriteModel{
+		WriteModel: eventstore.WriteModel{
+			ResourceOwner: resourceOwner,
+			InstanceID:    resourceOwner,
+		},
+		ids: ids,
+	}
+}
+
+func (wm *TargetsExistsWriteModel) Reduce() error {
+	for _, event := range wm.Events {
+		switch e := event.(type) {
+		case *target.AddedEvent:
+			if !slices.Contains(wm.existingIDs, e.ID) {
+				wm.existingIDs = append(wm.existingIDs, e.ID)
+			}
+		case *target.RemovedEvent:
+			i := slices.Index(wm.existingIDs, e.ID)
+			if i >= 0 {
+				wm.existingIDs = slices.Delete(wm.existingIDs, i, i)
+			}
```

Comment:
perhaps add a test case for the removed event?

Reply:
Same here